clippy_lints/src/ptr.rs

```diff
@@ -213,6 +213,21 @@ fn check_fn(cx: &LateContext, decl: &FnDecl, fn_id: NodeId, opt_body_id: Option<
                         },
                     );
                 }
+            } else if match_type(cx, ty, &paths::COW) {
+                let as_str = format!("{}", snippet_opt(cx, arg.span).unwrap());
```

Comment:
Is it ok to call unwrap() like this? I could import the function snippet() instead as that doesn't return a Result object

Reply:
I'll review more next week, but clippy dogfood will disallow the unwrap and the trivial format.

If there is no snippet, we should not be producing a suggestion.

Reply:
Interestingly the dogfood test passed for me locally with this code. 

I will update this code as you suggested, thanks.